src/system-core.js

```diff
@@ -283,26 +283,31 @@ function postOrderExec (loader, load, seen) {
       throw err;
     }
   });
-  if (depLoadPromises)
-    return Promise.all(depLoadPromises).then(doExec);
-
-  return doExec();
+  if (depLoadPromises) {
+    return load.E = Promise.all(depLoadPromises).then(doExec).finally(function() {
```

Comment:
This is quite a big algorithmic change, that will likely have repercussions. I think to be comfortable making these kinds of algorithmic changes it would help to see much more test cases being checked.